### ISSUE TITLE
ci: add workflow for updating compute IR flatbuffers files

### ISSUE THREAD

**Issue by cpcloud** [ibis-project/ibis#3079]
After #3045 is landed, the generated flatbuffers python code will be checked into the repo. We should add some CI infrastructure around this that:

1. Checks out the arrow repo
2. Generates the flatbuffers code
3. Pushes the changes in a commit to any PR that changes relevant (handwritten) files in ibis.

**Comment by cpcloud** [comment]
We're going to cut over to substrait, so this task is a wontfix.
